### PR TITLE
 MT-806 - Recsys Backend implementation - Mobile and FEW endpoints and performance improvements

### PR DESCRIPTION
**Pull request by adrian-musora**

### CODE REVIEW

File: src/Services/ContentService.php [hunk 1]
```diff
@@ -225,78 +227,183 @@ public function getById($id)
...
+            foreach($recommendations['recommendations'] as $index => $groupedBy) {
+                $recommendations['recommendations'][$index]['lessons'] = $this->getByIds($groupedBy['lessons']);
+            }
+            $content = $recommendations['recommendations'];
+            $content = Decorator::decorate($content, 'group');
+        } else {
+            $content = $this->getByIds($recommendations['recommendations']);
+        }
+        $filterOptions = [
+            "style" => ["All"],
```
Comment (roxanariza) [inline src/Services/ContentService.php]: Why we need the 'style','topic','instructor','artist'?

File: src/Services/ContentService.php [hunk 2]
```diff
@@ -225,78 +227,183 @@ public function getById($id)
...
+                $contentID != $value
+            )));
+        }
+        return $filteredContent;
+    }
+
+    private function getContentFilterResultsFromRecommendations($recommendations, $isGroupedBy)
+    {
+        if ($isGroupedBy) {
+            foreach($recommendations['recommendations'] as $index => $groupedBy) {
```
Comment (roxanariza) [inline src/Services/ContentService.php:346]: Just an idea for improvement :) Can we limit the number of lessons here? Only on All tab we have $isGroupedBy and on All tab we display 5 lessons for each group. If we keep the current flow, getByIds method from line 351 will pull all the lessons and all the lessons will be decorated with many decorators(a lot of unnecessary work for the BE)

File: src/Services/RecommendationService.php [hunk 3]
```diff
@@ -49,28 +64,52 @@ private function hasNoResults($brand, $section): bool
...
+        if (!isset($content[$userID])) {
+            $msg = print_r($content, true);
+            Log::warning("Malformed data from Huggingface: $msg");
+        }
+        return $content[$userID] ?? [];
+    }
+
+    private function getAllFilteredRecommendations($userID, $brand)
+    {
+        $url = env('HUGGINGFACE_URL');
```
Comment (roxanariza) [inline src/Services/RecommendationService.php]: Just a thought: the constants from the environment are stored in the config file, making it simpler to set them for tests and troubleshoot difficulties when the feature isn't functioning on your local

**Review by roxanariza — APPROVED** [review]
The code looks good, but will looks better if we limit the number of lessons :) Please see my comments!  Thank you!

File: src/Services/ContentService.php (hunk 1 above)
Comment (adrian-musora) [reply src/Services/ContentService.php]: Removed, pre-filter updates they were required for the FE or there was a crash IIRC.

File: src/Services/RecommendationService.php (hunk 3 above)
Comment (adrian-musora) [reply src/Services/RecommendationService.php]: I've moved them to the railconfig.php file and access them from there now. They can still be updated though the .env file if we need to quickly update the token or url for any reason.

File: src/Services/ContentService.php (hunk 2 above)
Comment (adrian-musora) [reply src/Services/ContentService.php:346]: That's a great idea. I've set the pagination default to 5 entries and updated the endpoint handlers acoordingly.

**Pull request merged**